src/core/ext/transport/chaotic_good/server/chaotic_good_server.cc

```diff
@@ -0,0 +1,370 @@
+      hpack_compressor_(std::make_unique<HPackCompressor>()),
+      hpack_parser_(std::make_unique<HPackParser>()),
+      connection_deadline_(Duration::Seconds(5)) {}
+
+ChaoticGoodServerListener::ActiveConnection::~ActiveConnection() {
+  listener_.reset();
+  if (receive_settings_activity_ != nullptr) receive_settings_activity_.reset();
+}
+
+void ChaoticGoodServerListener::ActiveConnection::Start(
+    std::unique_ptr<EventEngine::Endpoint> endpoint) {
+  GPR_ASSERT(handshaking_state_ == nullptr);
+  handshaking_state_ = std::make_shared<HandshakingState>(shared_from_this());
+  handshaking_state_->Start(std::move(endpoint));
+}
+
+void ChaoticGoodServerListener::ActiveConnection::GenerateConnectionID() {
+  std::string random_string;
+  int random_length = 8;
+  std::random_device rd;
+  std::mt19937 generator(rd());
+  std::string charset =
+      "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+  std::uniform_int_distribution<> distribution(0, charset.size() - 1);
+  random_string.reserve(8);
+  for (size_t i = 0; i < random_length; ++i) {
+    random_string += charset[distribution(generator)];
+  }
+  connection_id_ = Slice::FromCopiedString(random_string);
+}
+
+ChaoticGoodServerListener::ActiveConnection::HandshakingState::HandshakingState(
+    std::shared_ptr<ActiveConnection> connection)
+    : connection_(connection),
+      handshake_mgr_(std::make_shared<HandshakeManager>()),
+      arena_(connection_->arena_.get()),
+      context_(arena_.get()) {}
+
+void ChaoticGoodServerListener::ActiveConnection::HandshakingState::Start(
+    std::unique_ptr<EventEngine::Endpoint> endpoint) {
+  auto context = GetContext<Arena>();
+  GPR_ASSERT(context != nullptr);
+  handshake_mgr_->DoHandshake(
+      grpc_event_engine_endpoint_create(std::move(endpoint)),
+      connection_->args(), GetConnectionDeadline(connection_->args()), nullptr,
+      OnHandshakeDone, this);
+}
+void ChaoticGoodServerListener::ActiveConnection::HandshakingState::
+    OnHandshakeDone(void* arg, grpc_error_handle error) {
+  if (!error.ok()) {
+    gpr_log(GPR_ERROR, "Error server handshake failed: %s",
+            StatusToString(error).c_str());
+    return;
+  }
+  auto* args = static_cast<HandshakerArgs*>(arg);
+  GPR_ASSERT(args != nullptr);
+  if (args->endpoint == nullptr) {
+    gpr_log(GPR_ERROR, "Error server handshake done with null endpoint.");
+    return;
+  }
+  GPR_ASSERT(grpc_event_engine::experimental::grpc_is_event_engine_endpoint(
+      args->endpoint));
+  HandshakingState* self = static_cast<HandshakingState*>(args->user_data);
+  self->connection_->endpoint_ = std::make_shared<PromiseEndpoint>(
+      grpc_event_engine::experimental::grpc_take_wrapped_event_engine_endpoint(
+          args->endpoint),
+      SliceBuffer());
+  // Start receiving setting frames.
+  self->connection_->receive_settings_activity_ = OnReceive(self);
+}
+
+ActivityPtr
+ChaoticGoodServerListener::ActiveConnection::HandshakingState::OnReceive(
```

Comment:
Please add a TODO on where we expect to create the Chaotic Good transport

Reply:
Done, thanks for the suggestion!